src/sock/socket_session.cpp

```diff
@@ -28,6 +30,26 @@ void socket_session<T>::set_message_max_size(std::uint64_t size)
     ws.read_message_max(size);
 }
 
+/**
+ * Set thresholds to the socket session
+*/
+template <class T>
+void socket_session<T>::set_threshold(util::SESSION_THRESHOLDS threshold, std::uint64_t threshold_limit, std::uint64_t interval_in_ms)
+{
+
+    auto iterator = thresholds.find(threshold);
+    session_thresholds sess_threshold;
```

Comment:
Don't create a new instance of `session_threshold` every time you set a value. You have to update existing threshold struct or create new one if not exist.

Reply:
Used a wrong iterataor earlier which returned a read only pointer to the values in the unordered map so had to create a new object. It is resolved now